### DISCUSSION TITLE
thanks for rootfs encryption! ... and a small suggestion or two

### DISCUSSION THREAD

**Discussion by tinker2much** [gitbls/sdm#164] [Ideas]
First, thanks for rootfs encryption! I had been wanting to encrypt my pi5 desktop machine, but I had no idea how to go about it. I happened to look at the recent changes to sdm, and there it was. I pulled the latest sdm, read your doc about Disk Encryption, managed to find a big enough unused drive for use as a scratch drive, and off I went to do standalone encryption. It worked great. Your detailed documentation was just right - very informative - it told me what to do, what to expect, what to be aware of, and what to be careful about. Very well done. 

A few tiny suggestions: 

1. provide some estimates of the time steps will take (even if very rough). I had no idea and that was a little worrying. In my case, I had just under 40GB of data, and with a Kingston A400 as the existing drive, and a PNY 64GB USB stick as the scratch drive, the shrinking took 5 minutes, the copy to sdb took 42 minutes, and the copy back to the newly encrypted sda took 23 minutes. Could you call that roughly 1GB per minute on the copy steps? 

2. When you "print the size of the rootfs", its the result of the shrink command I guess? That showed up as "9586750 4k blocks", which I had to grab a calculator to turn into 39GB. I don't know how hard it would be to capture that result and translate it to human readable, but that would be much more informative. 

3. when it told me at the end that sda finally took up "117076692 4k blocks", off the top of my head, I wondered if that was the size of the encrypted data or the size of the drive. Once converted to 468GB, I knew it was simply the size of the drive. df -h tells me I now have 37GB used. Human readable is always better - who thinks in terms of 4k blocks?   ;-)

**Comment by gitbls** [comment]
Thanks for the feedback! I was hoping that someone would give it a go for confirmation that it actually works. ;)

Regarding your suggestions:
1. Looking into how to do this with reasonably reliable estimates for the various combinations: SSD<>SSD, SSD<>USB, USB<>USB
2. Yes, will do this
3. The number that's printed is from `resize2fs`, so may have to do a separate message, or perhaps just do a `df -H` after the shrink. Looking into this.
